### PR TITLE
Move STDIO input reads off the cooperative executor

### PR DESCRIPTION
**Pull request by lynnswap**
## Purpose

Fix the main-branch CI hang observed in [run 29402610133](https://github.com/lynnswap/XcodeMCPKit/actions/runs/29402610133/job/87310906177).

The apparent progress-test stall was buffered output. Under a single cooperative worker, `FileHandle.AsyncBytes` performed a blocking stdin read and starved the task responsible for writing and closing the adapter pipe.

## Changes

- move STDIO input ownership to a dedicated `DispatchIO` channel
- bound read-ahead to one 64 KiB operation while preserving input order and backpressure
- close only the duplicated input descriptor and preserve the caller-owned `FileHandle`
- preserve graceful EOF draining, immediate read-error aborts, cancellation, and descriptor-cleanup completion
- add event-driven tests for bounded reads, exact-boundary EOF, active-read cancellation, read failures, deinitialization, and caller ownership
- use an injected clock to prove read failures do not enter the EOF drain loop
- bound the test event observation so regressions fail instead of hanging CI; elapsed time is not used to establish behavior

## Testing

- `./scripts/check.sh`
  - 956 package tests passed
  - 24 process runtime tests passed
  - 13 STDIO adapter tests passed
- strict single-worker reproduction passed with `LIBDISPATCH_COOPERATIVE_POOL_STRICT=1` (14 focused tests)
- branch-wide `codex-review` against pinned `main@d2b7e76d`: no findings

### CODE REVIEW

**Comment by lynnswap** [comment]
@codex review

File: Sources/XcodeMCPProxyKit/Stdio/StdioInputChannel.swift [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+        let terminal = AsyncTerminalSignal()
+        self.terminal = terminal
+        self.channel = DispatchIO(
+            type: .stream,
+            fileDescriptor: descriptor,
+            queue: callbackQueue
+        ) { _ in
+            _ = Darwin.close(descriptor)
+            try? handle.close()
```
[the remark this answers is not in the document]
Comment (lynnswap) [reply Sources/XcodeMCPProxyKit/Stdio/StdioInputChannel.swift]: Fixed in 3b41ed25. `StdioInputChannel` now closes only its duplicated descriptor and no longer captures or closes the caller-owned `FileHandle`. `stopBeforeReadLeavesCallerHandleOpenAndIsIdempotent` verifies that the caller descriptor remains open after channel cleanup.

File: Tests/ProxyStdioAdapterTests/StdioAdapterIntegrationTests.swift [hunk 2]
```diff
@@ -87,6 +85,46 @@ struct StdioAdapterContractTests {
...
+        let adapter = StdioAdapter(
+            requestTimeout: nil,
+            inputReader: inputReader,
+            output: outputPipe.fileHandleForWriting,
+            recipe: MCPTransportRecipe { client },
+            shutdownPolicy: StdioAdapterShutdownPolicy(clock: clock)
+        )
+
+        try await adapter.start()
+        #expect(try await client.nextSentBody(at: 2) == Data(toolsListRequest.utf8))
```
[the remark this answers is not in the document]
Comment (lynnswap) [reply Tests/ProxyStdioAdapterTests/StdioAdapterIntegrationTests.swift]: Fixed in 3b41ed25. The test now uses the existing bounded `sentBody(at:)` waiter, so a stuck send fails instead of hanging CI. Event delivery remains the behavioral synchronization, while the injected clock continues to control the drain semantics.

**Review by lynnswap — COMMENTED** [review]
@codex review
